### ISSUE TITLE
Chrome launch doesn't work properly with node/protractor under WSL

### ISSUE THREAD

**Issue by rfgamaral** [Microsoft/vscode#49908]
- VSCode Version: 123.1
- OS Version: Windows 10 (Version 1803, Build 17134.1)

**Steps to Reproduce:**

I have launch configuration like this:

```json
{
    "type": "node",
    "request": "launch",
    "name": "Debug Local Tests",
    "cwd": "${workspaceFolder:my-project}",
    "program": "${workspaceFolder:my-project}/node_modules/protractor/bin/protractor",
    "args": [
        "tests/config/protractor/protractor.conf.override.js"
    ],
    "runtimeExecutable": "~/.node/bin/node",
    "useWSL": true
}
```

This is the log I get when I run this task:

```
C:\WINDOWS\System32\bash.exe -ic '~/.node/bin/node --inspect-brk=47942 node_modules/protractor/bin/protractor tests/config/protractor/protractor.conf.override.js' 
Debugger listening on ws://127.0.0.1:47942/04839764-46ff-4801-b555-43f6d54ce2c6
Debugger attached.
Report destination:   reports/htmlScreenshotReporter/testsReport.html
(node:3) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
[15:42:18] I/launcher - Running 1 instances of WebDriver
[15:42:18] I/local - Starting selenium standalone server...
[15:42:20] I/local - Selenium standalone server started at http://172.26.213.33:52848/wd/hub
```

And I also get this warning:

![image](https://user-images.githubusercontent.com/96476/40064139-d019328c-5856-11e8-8d48-c338395f5f12.png)

After clicking "OK", Chrome launches with an empty tab and does nothing. Eventually I get this exception inside Code:

```
Exception has occurred: WebDriverError
WebDriverError: unknown error: Chrome failed to start: exited abnormally
  (Driver info: chromedriver=2.38.552522 (437e6fbedfa8762dec75e2c5b3ddb86763dc9dcb),platform=Linux 4.4.0-17134-Microsoft x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 60.68 seconds
Build info: version: '3.12.0', revision: '7c6e0b3', time: '2018-05-08T15:15:08.936Z'
System info: host: 'LP5J44QH2', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.4.0-17134-Microsoft', java.version: '10.0.1'
Driver info: driver.version: unknown
    at Object.checkLegacyResponse (C:\Workspace\my-project\node_modules\selenium-webdriver\lib\error.js:546:16)
    at parseHttpResponse (C:\Workspace\my-project\node_modules\selenium-webdriver\lib\http.js:509:14)
    at doSend.then.response (C:\Workspace\my-project\node_modules\selenium-webdriver\lib\http.js:441:31)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
From: Task: WebDriver.createSession()
    at Function.createSession (C:\Workspace\my-project\node_modules\selenium-webdriver\lib\webdriver.js:769:25)
    at Function.createSession (C:\Workspace\my-project\node_modules\selenium-webdriver\chrome.js:761:16)
    at createDriver (C:\Workspace\my-project\node_modules\selenium-webdriver\index.js:170:34)
    at Builder.build (C:\Workspace\my-project\node_modules\selenium-webdriver\index.js:626:17)
    at Local.getNewDriver (C:\Workspace\my-project\node_modules\protractor\built\driverProviders\driverProvider.js:53:34)
    at Runner.createBrowser (C:\Workspace\my-project\node_modules\protractor\built\runner.js:195:44)
    at q.then.then (C:\Workspace\my-project\node_modules\protractor\built\runner.js:339:30)
    at _fulfilled (C:\Workspace\my-project\node_modules\protractor\node_modules\q\q.js:834:55)
    at self.promiseDispatch.done (C:\Workspace\my-project\node_modules\protractor\node_modules\q\q.js:863:31)
    at Promise.promise.promiseDispatch (C:\Workspace\my-project\node_modules\protractor\node_modules\q\q.js:796:14)
```

Help figuring out this issue is appreciated :)

**Comment by rfgamaral** [comment]
I believe there's two different issues here:

1) The first one is that Chrome is being launcher from WSL thinking it's running under WSL and not Windows. Thus is trying to create a temporary folder under `/tmp` and not using my Windows profile data folder. Although Chrome still runs, it's an annoying message and I don't know if this could cause other issues.
2) Chrome is a Windows application running on Windows but the Webdriver is a linux binary. The linux binary was installed because this project has an `npm` task to install/update the Webdriver, which was detected as a Linux system since it was running under WSL.

I'm not sure if these issues should be here or https://github.com/Microsoft/WSL.

**Comment by rfgamaral** [comment]
Forget about 2 for a second, let's focus on 1.

I expect Chrome to launch successfully without errors/warnings. Currently I get the warning above. Outside of VSCode I launch it directly from a Windows start menu shortcut.

**Comment by rfgamaral** [comment]
@roblourens I've solved all my problems with manual Protractor configuration.

The first problem was solved by pointing the `config.capabilities.chromeOptions.binary` Protractor configuration option from `/mnt/c/Users/AmaralR/AppData/Local/Google/Chrome SxS/Application/chrome.exe` to `C:/Users/AmaralR/AppData/Local/Google/Chrome SxS/Application/chrome.exe`. If I use the first form (`/mtn/c/...`), I get the the read/write error I posted initially. This is an option I always had to manual configure before using Code/WSL anyways.

Once I figured out how to fix the first issue, the second one was a bit obvious. Chrome is running as a Windows application and as such I need the appropriate Windows Selenium chrome webdriver and not the Linux version which was being installed because of angular/webdriver-manager#310 and/or Microsoft/WSL#3198. To solve that, I manually download Selenium Server Standalone and the Chrome webdriver and added `config.chromeDriver = '/mnt/c/Selenium/chromedriver.exe';` and `config.seleniumServerJar = '/mnt/c/Selenium/selenium-server-standalone-3.12.0.jar';` to Protractor configuration. Yes, both these options need to use `/mnt/c/...` instead of `C:/...`.

The only thing I'm confused by is the different usages of `/mnt/c/...` and `C:/...`.

So, with this launch configuration:

```json
{
    "type": "node",
    "request": "launch",
    "name": "Debug Local Tests",
    "cwd": "${workspaceFolder:my-project}",
    "program": "${workspaceFolder:my-project}/node_modules/protractor/bin/protractor",
    "args": [
        "tests/config/protractor/protractor.conf.override.js"
    ],
    "runtimeExecutable": "~/.node/bin/node",
    "useWSL": true
}
```

And this Protractor configuration:

```js
const config = require('./protractor.conf.js').config;

config.chromeDriver = '/mnt/c/Selenium/chromedriver.exe';

config.seleniumAddress = null;
config.seleniumServerJar = '/mnt/c/Selenium/selenium-server-standalone-3.12.0.jar';

config.capabilities = {
    browserName: 'chrome',
    chromeOptions: {
        binary: 'C:/Users/AmaralR/AppData/Local/Google/Chrome SxS/Application/chrome.exe'
    },
    shardTestFiles: false,
    maxInstances: 1
};

exports.config = config;
```

I can run my automation tests under WSL through VS Code.

I guess this really has nothing to do with VS Code, that configuration is fine. :)
